openshift-kube-apiserver/admission/customresourcevalidation/authentication/validate_authentication.go

```diff
@@ -25,7 +29,12 @@ func Register(plugins *admission.Plugins) {
 				configv1.GroupVersion.WithResource("authentications").GroupResource(): true,
 			},
 			map[schema.GroupVersionKind]crvalidation.ObjectValidator{
-				configv1.GroupVersion.WithKind("Authentication"): authenticationV1{},
+				configv1.GroupVersion.WithKind("Authentication"): authenticationV1{
+					cel: celStore{
+						compiler:                   authenticationcel.NewDefaultCompiler(),
+						knownCompilableExpressions: sets.New[string](),
```

Comment:
How will you handle concurrent access to this map?

Reply:
Ah, forgot to add a mutex - good callout

Reply:
You might use https://pkg.go.dev/k8s.io/utils/lru#Cache which is safe for concurrent access and handles eviction for you.